pkg/metrics/listener.go

```diff
@@ -30,9 +30,11 @@ func NewHTTPListener(addr string) error {
 
 	log.L.Infof("Start metrics HTTP server on %s", addr)
 
-	if err := http.ListenAndServe(addr, nil); err != nil {
-		return fmt.Errorf("error serve on %s: %v", addr, err)
-	}
+	go func() {
+		if err := http.ListenAndServe(addr, nil); err != nil {
+			log.L.Errorf("error serve on %s: %v", addr, err)
```

Comment:
Can we make sure to exit snapshotter once `http.ListenAndServe` fails?

Reply:
Better not to, we should try to maintain the snapshots service , I suppose.

Reply:
So the user can only determine if the metrics server is working by checking the error log? I think it's a not good design.

Reply:
Um. If we indeed need to ensure that the metrics server is successfully started, perhaps we can try to connect to the listener in the main goroutine

Reply:
I think we just need to move the connection `listen` out from `ListenAndServe`. `listen` is likely to fail when the port is  already bound when serve can only fail when the connection is closed by accident

Reply:
Agree. It is sufficient to quit `snapshotter` if the port is already bound.